src/mesh/high_order_grid.cpp

```diff
@@ -1050,6 +1050,49 @@ void HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::execute_coarsen
     //}
 }
 
+template <int dim, typename real, typename MeshType, typename VectorType, typename DoFHandlerType>
+dealii::Point<dim> HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::smallest_cell_coordinates()
+{
+    const int iproc = dealii::Utilities::MPI::this_mpi_process(mpi_communicator);
+    const dealii::Point<dim> unit_vertex = dealii::GeometryInfo<dim>::unit_cell_vertex(0);
+    double min_diameter_local;
+    double current_cell_diameter;
+    auto cell = dof_handler_grid.begin_active();
+    auto endcell = dof_handler_grid.end();
+    min_diameter_local = cell->diameter();
+    dealii::Point<dim> smallest_cell_coord; 
+    for (; cell!=endcell; ++cell) 
+    {
+        current_cell_diameter = cell->diameter();
+        if ((min_diameter_local > current_cell_diameter) && (cell->is_locally_owned()))
+        {
+            min_diameter_local = current_cell_diameter;
+            smallest_cell_coord = mapping_fe_field->transform_unit_to_real_cell(cell, unit_vertex);
+            //smallest_cell_coord = cell->center();
+        }
+    }
+    
+    MPI_Barrier(mpi_communicator); // wait till all processors are done
+    dealii::Utilities::MPI::MinMaxAvg minindexstore;
+    minindexstore = dealii::Utilities::MPI::min_max_avg(min_diameter_local, mpi_communicator); 
+
+    int n_proc_small = minindexstore.min_index; // Processor containing smallest diameter
+    dealii::Point<dim> smallest_cell_global; // Initialized with 0
+
+    if (iproc == n_proc_small)
+    {
+        smallest_cell_global = smallest_cell_coord;         
+    }
+    
+    MPI_Barrier(mpi_communicator); // wait for all processors
+
+    for (int i=0; i<dim; i++)
+        smallest_cell_global[i] = dealii::Utilities::MPI::sum(smallest_cell_global[i],mpi_communicator); // Update values in all processors
```

Comment:
Look into using MPI_Bcast. While sum might work, it's not the correct operation that tells the reader of the code what you're doing. Relevant tutorials:
https://mpitutorial.com/tutorials/mpi-broadcast-and-collective-communication/
https://mpitutorial.com/tutorials/mpi-scatter-gather-and-allgather/

Reply:
Thanks for the links. This function now uses MPI_Bcast.